cg/meta/clean/clean_retrieved_spring_files.py

```diff
@@ -0,0 +1,43 @@
+import datetime
+import logging
+from pathlib import Path
+
+from housekeeper.store.models import File
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+
+LOG = logging.getLogger(__name__)
+
+
+class CleanRetrievedSpringFilesAPI:
+    """API for cleaning archived Spring files which have been retrieved."""
+
+    def __init__(self, housekeeper_api: HousekeeperAPI, dry_run: bool):
+        self.housekeeper_api: HousekeeperAPI = housekeeper_api
+        self.dry_run = dry_run
+
+    def _get_files_to_remove(self, days_since_retrieval: int) -> list[File]:
+        """Returns all Spring files which were retrieved more than given amount of days ago."""
+        return self.housekeeper_api.get_spring_files_retrieved_before(
+            date=datetime.datetime.now() - datetime.timedelta(days=days_since_retrieval)
```

Comment:
`from datetime import datetime, timedelta`
```suggestion
            date=datetime.now() - timedelta(days=days_since_retrieval)
```